expose runner's env to prelaunch command script

eikeno opened the pull request:
By adding `env' dictionnary passed to the process a new key: LUTRIS_RUNNER_SYSTEM_CONFIG
containing json dump of the the dict: `self.runner.system_config'

Implements feature request: #5550

eikeno committed:
expose runner's env to prelaunch command script

By adding `env' dictionnary passed to the process a new key:
LUTRIS_RUNNER_SYSTEM_CONFIG
containing json dump of the the dict: `self.runner.system_config'

Implements feature request: #5550

danieljohnson2 commented:
I'm of two minds on this feature. That's a possibly big chunk of JSON you are dumping into the environment. On the other hand, it's just the ``system_config`` and you might want the ``running_config`` or the ``game_config``.

So here's my idea. Can we have additional options right next to the environment list box that would enable ``LUTRIS_RUNNER_SYSTEM_CONFIG`` and two more for ``LUTRIS_RUNNER_RUNNER_CONFIG`` or ``LUTRIS_RUNNER_GAME_CONFIG``?

eikeno commented:
yes that sounds feasible  without too much complication. I'll check this and update

eikeno committed:
Make exposition of environment optional and cover all scripts

Scripts covered: pre, post & manual

Exposition is now individually configurable in options, under
Env vars section, to control these env. variables:
LUTRIS_RUNNER_SYSTEM_CONFIG
LUTRIS_RUNNER_RUNNER_CONFIG
LUTRIS_RUNNER_GAME_CONFIG

env manipulation moved to:
lutris.util.system.expose_environment
as it seems appropriate and avoids dedundancy.
Could be moved anywhere else as it's a static method.

Had to use "Any" type hinting for the runner parameter, as being more specific
causes circular import errors (for LutrisConfig or Runner) at run time.

Version currently used of Mypy is OK with that, may need
revision when we upgrade Mypy in the future.

danieljohnson2 commented:
I like it!

eikeno committed:
Merge branch 'lutris:master' into export_prelaunch

eikeno committed:
Merge branch 'lutris:master' into export_prelaunch
